src/Xamarin.Android.Build.Tasks/Tasks/JavaToolTask.cs

```diff
@@ -93,13 +100,16 @@ at com.android.dx.command.Main.main(Main.java:106)
 		protected override bool HandleTaskExecutionErrors ()
 		{
 			if (foundError) {
+				AssemblyIdentityMap assemblyMap = new AssemblyIdentityMap ();
+				assemblyMap.Load (AssemblyIdentityMapFile);
 				errorText.Clear ();
+				mappingText.Clear ();
 				foreach (var line in errorLines) {
-					if (!ProcessOutput (line))
+					if (!ProcessOutput (line, assemblyMap))
 						break;
 				}
 				if (foundError && errorText.Length > 0) {
-					Log.LogError (ToolName, DefaultErrorCode, null, file, line - 1, column + 1, 0, 0, errorText.ToString ());
+					Log.LogError (ToolName, DefaultErrorCode, null, file, line - 1, column + 1, 0, 0, errorText.ToString () + String.Join ("\n", mappingText));
```

Comment:
I wonder if this should be `string.Join (Environment.NewLine, …)` instead of `string.Join ("\n", …)`, and *that* is why the [error messages loses all line endings](https://github.com/xamarin/xamarin-android/pull/7012#issuecomment-1139173735).